third_party/blink/renderer/modules/indexeddb/idb_database.cc

```diff
@@ -114,8 +114,13 @@ IDBDatabase::IDBDatabase(
 }
 
 IDBDatabase::~IDBDatabase() {
-  if (!close_pending_ && backend_)
-    backend_->Close();
+  if (!close_pending_ && backend_) {
+    // Explicitly leak the database proxy, as we are likely in a GC, and
+    // closing will issue IPC messages that need to be recorded.
+    if (!recordreplay::IsRecordingOrReplaying()) {
```

Comment:
Nit: pass a string argument to `IsRecordingOrReplaying()` to allow us to disable this particular behaviour using a runtime environment variable.  e.g.:

`recordreplay::IsRecordingOrReplaying("IDBDatabase::~IDBDatabase")`